### ISSUE TITLE
Auto-import fails for default exports that are the result of declaration merging

### ISSUE THREAD

**Issue by dfreeman** [Microsoft/TypeScript#24535]
**TypeScript Version:**  3.0.0-dev.20180531

**Search Terms:** various combinations of `merge`, `default`, `import`, `suggestion`, `auto`

**Code**

This issue occurs when a default export is augmented in another declaration, which I originally ran into when testing out https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26154

```ts
// file: a.ts
declare module 'my-class' {
    export default class MyClass { /* ... */ }
}

// file: b.ts
declare module 'my-class' {
    export default interface MyClass { /* ... */ }
}
```

**Expected behavior:**

Typing `MyClass` and requesting suggested fixes includes the option to auto-import `MyClass from 'my-class'`.

**Actual behavior:**

An entry in `MyClass` appears in the list, but selecting it does nothing. Debugging indicates that [this assertion](https://github.com/Microsoft/TypeScript/blob/e9dda76874235a8b638efafff2d52e12ae6c0714/src/services/codefixes/importFixes.ts#L99) is failing because the `exportInfos` array is empty.

I have a branch with a new (initially failing) unit test and a fix, but wanted to get a bug filed first.

**Playground Link:** N/A (as far as I can tell, suggested fixes aren't available in the playground)

**Related Issues:** I found https://github.com/Microsoft/TypeScript/issues/14080 in the course of searching around, but parallel `export default`s appear to work fine for augmentation at least for the use cases I've run into.

**Comment by mhegazy** [comment]
This is just not a supported pattern at the moment. you can not augment a default export. https://github.com/Microsoft/TypeScript/issues/14080 tracks fixing that. i suggests adding this repro to https://github.com/Microsoft/TypeScript/issues/14080

**Comment by dfreeman** [comment]
I'm a bit confused — the augmentation itself appears to work exactly as expected, e.g. in [this playground](http://www.typescriptlang.org/play/#src=declare%20module%20'm'%20%7B%0D%0A%20%20export%20default%20class%20MyClass%20%7B%0D%0A%20%20%20%20a%3A%20string%3B%0D%0A%20%20%7D%0D%0A%7D%0D%0A%0D%0Adeclare%20module%20'm'%20%7B%0D%0A%20%20export%20default%20interface%20MyClass%20%7B%0D%0A%20%20%20%20b%3A%20number%3B%0D%0A%20%20%7D%0D%0A%7D%0D%0A%0D%0Adeclare%20module%20'foo'%20%7B%0D%0A%20%20import%20M%20from%20'm'%3B%0D%0A%0D%0A%20%20const%20instance%3A%20M%3B%0D%0A%20%20type%20a%20%3D%20typeof%20instance.a%3B%0D%0A%20%20type%20b%20%3D%20typeof%20instance.b%3B%0D%0A%7D). The only problem I was running into was with the auto-import.

**Comment by mhegazy** [comment]
looks like we fixed #14080 at some point..
